tests/financial/test_national_stock_exchange_of_india.py

```diff
@@ -0,0 +1,742 @@
+            "2027-09-04",
+            "2028-08-23",
+            "2029-09-11",
+            "2030-09-01",
+            "2031-09-20",
+            "2032-09-08",
+            "2033-08-28",
+            "2034-09-16",
+            "2035-09-05",
+        )
+        self.assertNonObservedHolidayName(name, dt)
+        self.assertNonObservedHolidayName(name, range(2001, 2036))
+
+    def test_guru_nanak_jayanti(self):
+        name = "Guru Nanak Jayanti"
+        dt = (
+            "2001-11-30",
+            "2002-11-19",
+            "2003-11-08",
+            "2004-11-26",
+            "2005-11-15",
+            "2006-11-05",
+            "2007-11-24",
+            "2008-11-13",
+            "2009-11-02",
+            "2010-11-21",
+            "2011-11-10",
+            "2012-11-28",
+            "2013-11-17",
+            "2014-11-06",
+            "2015-11-25",
+            "2016-11-14",
+            "2017-11-04",
+            "2018-11-23",
+            "2019-11-12",
+            "2020-11-30",
+            "2021-11-19",
+            "2022-11-08",
+            "2023-11-27",
+            "2024-11-15",
+            "2025-11-05",
+            "2027-11-14",
+            "2028-11-02",
+            "2029-11-21",
+            "2030-11-10",
+            "2031-11-28",
+            "2032-11-17",
+            "2033-11-06",
+            "2034-11-25",
+            "2035-11-15",
+        )
+        self.assertNonObservedHolidayName(name, dt)
+        self.assertNonObservedHolidayName(name, range(2001, 2026))
+        self.assertNonObservedHolidayName(name, range(2027, 2036))
+
+    def test_muharram(self):
+        name = "Muharram (estimated)"
+        dt = (
+            "2001-04-04",
+            "2002-03-24",
+            "2003-03-13",
+            "2004-03-01",
+            "2005-02-19",
+            "2006-02-09",
+            "2007-01-29",
+            "2008-01-19",
+            "2009-12-27",
+            "2010-12-16",
+            "2011-12-05",
+            "2012-11-24",
+            "2013-11-13",
+            "2014-11-03",
+            "2015-10-23",
+            "2016-10-11",
+            "2017-09-30",
+            "2018-09-20",
+            "2019-09-09",
+            "2020-08-29",
+            "2021-08-18",
+            "2026-06-25",
+            "2027-06-15",
+            "2028-06-03",
+            "2029-05-23",
+            "2030-05-12",
+            "2031-05-02",
+            "2032-04-20",
+            "2033-04-10",
+            "2034-03-30",
+            "2035-03-20",
+        )
+        self.assertNonObservedHolidayName(name, dt)
+        self.assertNonObservedHolidayName(name, range(2001, 2022))
+        self.assertNonObservedHolidayName(name, range(2026, 2036))
+
+        name = "Muharram"
+        dt = (
+            "2022-08-09",
+            "2023-07-29",
+            "2024-07-17",
+            "2025-07-06",
+        )
+        self.assertNonObservedHolidayName(name, dt)
+        self.assertNonObservedHolidayName(name, range(2022, 2026))
```

Comment:
```suggestion
        name = "Muharram"
        self.assertNonObservedHolidayName(
            name,
            "2022-08-09",
            "2023-07-29",
            "2024-07-17",
            "2025-07-06",
        )
        self.assertNonObservedHolidayName(
            name, self.nonobs_no_estimated_holidays, range(2001, 2050)
        )
```
And the same for rest Islamic holidays.